content/docs/rfcs/13/README.md

```diff
@@ -59,139 +59,262 @@ The following are not considered as part of the adversarial model:
 In specific, the communication channels are assumed to be secure.
 
 # Wire Specification
-Peers communicate with each other using a request / response API. 
-The messages sent are Protobuf RPC messages which are implemented using [protocol buffers v3](https://developers.google.com/protocol-buffers/).
-The following are the specifications of the Protobuf messages.
 
 ## Payloads
 
 ```protobuf
 syntax = "proto3";
 
-message Index {
-  bytes digest = 1;
-  sint64 receiverTime = 2;
-  sint64 senderTime = 3;
-  string pubsubTopic = 4;
-}
-
-message PagingInfo {
-  uint64 pageSize = 1;
-  Index cursor = 2;
-  enum Direction {
-    BACKWARD = 0;
-    FORWARD = 1;
-  }
-  Direction direction = 3;
-}
+// Protocol identifier: /vac/waku/store-query/3.0.0
+package waku.store.v3;
 
-message ContentFilter {
-  string contentTopic = 1;
-}
+import "waku/message/v1/message.proto";
 
-message HistoryQuery {
-  // the first field is reserved for future use
-  string pubsubtopic = 2;
-  repeated ContentFilter contentFilters = 3;
-  PagingInfo pagingInfo = 4;
+message WakuMessageKeyValue {
+  optional bytes message_hash = 1; // Globally unique key for a Waku Message
+  optional waku.message.v1.WakuMessage message = 2; // Full message content as value
```

Comment:
When querying with hashes, there is no way to know which shard a message was sent on. `WakuMessage` does not contain the pubsub topic.

Reply:
I'd vote for it to be part of the `WakuMessageKeyValue` so the `WakuMessage` protobuffer is not modified!

Reply:
Good points. We indeed need the `pubsub_topic`. @richard-ramos @SionoiS @chaitanyaprem please take a look at https://github.com/vacp2p/rfc/pull/665/commits/5d69ab58b34d32af99e60b109ad784ad1dd9e345 and see if you agree with the addition.

Reply:
Looks good. 
Just wondering if we should return pubsubTopic or cluster and shard since anyways named sharding would be deprecated.

Reply:
Not a strong opinion about this, but when we previously discussed this the intuition was:
- Waku core protocols still operate on the idea of pubsub-topic (i.e. string) defined subnetworks, as a reflection of its libp2p base
- the string formatting we provide for cluster and shard is run on top of this as a type of application, but it does not change the fundamental network definition

Of course, part of keeping it `string` is to remain consistent with filter, relay, lightpush for now too and avoid the potential inconsistency in cross-protocol composition. Perhaps if we do upgrade protocols to cluster-shard we should do all protocols at once (i.e. not now... :) )?

Reply:
Agreed, better to upgrade all protos in one shot rather than partly. This will avoid confusion as well.